integration-tests/contracts/ValueCalls.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+// @unsupported: evm
+pragma solidity >=0.7.0;
+
+contract ValueContext {
+    function getSelfBalance() external view returns(uint256) {
+        uint selfBalance;
```

Comment:
Please add an additional test where `getSelfBalance` is *payable* and show that the result is `balanceBefore + msg.value`.

Reply:
added